uqcsbot/api.py

```diff
@@ -67,17 +68,30 @@ def __call__(self, **kwargs) -> dict:
         If the APIMethodProxy was constructed with `is_async=True` runs the
         request asynchronously via:
             asyncio.get_event_loop().run_in_executor(None, req)
+
+        Attempts to retry the API call if rate-limited.
         """
         fn = partial(
             self._client.api_call,
             self._method,
             **kwargs
         )
-        if self._async:
-            loop = asyncio.get_event_loop()
-            return loop.run_in_executor(None, fn)
+        retry_count = 0
+        while retry_count < 5:
+            if self._async:
+                loop = asyncio.get_event_loop()
+                result = loop.run_in_executor(None, fn)
+            else:
+                result = fn()
+            if not result['ok'] and result['error'] == 'ratelimited':
```

Comment:
Can't just use result like this - needs to be awaited if `self._async`

Reply:
like, `result = await loop.run_in_executor(None, fn)`?

Reply:
like, `result = await loop.run_in_executor(None, fn)`?